vault/resource_ssh_secret_backend_role_test.go

```diff
@@ -112,12 +133,39 @@ func TestAccSSHSecretBackendRole(t *testing.T) {
 				),
 			},
 			{
-				ResourceName:      resourceName,
-				ImportState:       true,
-				ImportStateVerify: true,
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: importStateVerifyIgnore,
 			},
-		},
+		}
+	}
+
+	t.Run("vault-1.11-and-below", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			Providers: testProviders,
+			PreCheck: func() {
+				testutil.TestAccPreCheck(t)
+				SkipIfAPIVersionGTE(t, testProvider.Meta(), provider.VaultVersion112)
+
+			},
+			CheckDestroy: testAccSSHSecretBackendRoleCheckDestroy,
+			Steps:        getSteps([]string{"allowed_domains_template"}),
```

Comment:
Why do we need to ignore `allowed_domains_template` here?

Reply:
Without it, I was getting a test failure:
```
    resource_ssh_secret_backend_role_test.go:145: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=1) {
         (string) (len=24) "allowed_domains_template": (string) (len=4) "true"
        }
```
and I think it was easier to ignore the field for vault 1.11 than to create another config that excluded it

Reply:
I think this means that the checks are wrong since the "actual" (top) is what we expect to happen since this field does not exist in 1.11

Reply:
gotcha! @vinay-gopalan helped clarify this for me, and I think it makes sense. Updated to no longer ignore fields, but rather use a dynamic config instead.